version-catalog/src/main/kotlin/io/spine/internal/catalog/entry/CatalogEntry.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.catalog.entry
+
+import io.spine.internal.catalog.Alias
+import io.spine.internal.catalog.CatalogItemNotation
+import io.spine.internal.catalog.CatalogRecord
+
+/**
+ * A base catalog entry.
+ *
+ * The main idea behind the concept of entries is to provide a declarative way
+ * to define version catalog items. Entries expose a declarative API, leaving
+ * behind the scene all imperative code.
+ *
+ * Only object declarations are meant to serve as concrete entries.
+ *
+ * As a base entry, this class takes a responsibility for the following:
+ *
+ *  1. Support of nesting. An entry can nest one or more other entries.
+ *  2. Automatic aliasing for entries. An alias is picked up from the object's
+ *     name, taking into account its nesting.
+ *
+ * As the ultimate goal of its existence, the class provides [allRecords] method.
```

Comment:
Let's say less in general, but more in specific. This sentence says really nothing. You can easily remove it altogether.